src/definitions.ts

```diff
@@ -5,6 +5,14 @@ export interface GenericOAuth2Plugin {
    * @returns {Promise<any>} the resource url response
    */
   authenticate(options: OAuth2AuthenticateOptions): Promise<any>;
+  /**
+   * Listens for OAuth implicit redirect flow queryString CODE to generate an access_token
+   * @param {OAuth2RedirectAuthenticationOptions} options
+   * @returns {Promise<any>} the token endpoint response
+   */
+  redirectFlowCodeListener(
+    options: OAuth2RedirectAuthenticationOptions,
```

Comment:
Could you rename `OAuth2RedirectAuthenticationOptions` to `ImplicitFlowRedirectOptions`